internal/engine/docker_compose_log_manager.go

```diff
@@ -0,0 +1,148 @@
+package engine
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/windmilleng/tilt/internal/dockercompose"
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+// Collects logs from running docker-compose services.
+type DockerComposeLogManager struct {
+	watches map[model.ManifestName]DockerComposeLogWatch
+}
+
+func NewDockerComposeLogManager() *DockerComposeLogManager {
+	return &DockerComposeLogManager{
+		watches: make(map[model.ManifestName]DockerComposeLogWatch),
+	}
+}
+
+// Diff the current watches against set of current docker-compose services, i.e.
+// what we SHOULD be watching, returning the changes we need to make.
+func (m *DockerComposeLogManager) diff(ctx context.Context, st store.RStore) (setup []DockerComposeLogWatch, teardown []DockerComposeLogWatch) {
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	// If we're not watching the mounts, then don't bother watching logs.
+	if !state.WatchMounts {
+		return nil, nil
+	}
+
+	for _, ms := range state.ManifestStates {
+		if !ms.Manifest.IsDockerCompose() {
+			continue
+		}
+
+		existing, isActive := m.watches[ms.Manifest.Name]
```

Comment:
does `isActive` here mean that a watch is active if the `DockerComposeLogWatch` in that map is not null?

Reply:
If so, how can it be null?